packages/components/src/Datalist/Datalist.component.js

```diff
@@ -0,0 +1,341 @@
+	 * @param event
+	 * @param value
+	 */
+	onChange(event, { value }) {
+		this.updateGroups(value, true);
+		this.updateValue(event, value, false);
+		// resetting selection here in order to reinit the section + item indexes
+		this.resetSelection();
+	}
+
+	/**
+	 * Display suggestions on focus
+	 */
+	onFocus(event) {
+		event.target.select();
+		this.updateGroups(this.state.value);
+		this.updateSelectedIndexes(this.state.value);
+	}
+
+	/**
+	 * Hook on input keydown
+	 * ESC: reset the value to the previous persited one
+	 * ENTER: select a suggestion, persist the value, or submit the form
+	 * UP/DOWN: update the active suggestion index
+	 * @param event The keydown event
+	 * @param highlightedItemIndex The previous focused suggestion index
+	 * @param newHighlightedItemIndex The new focused suggestion index
+	 */
+	onKeyDown(event, params) {
+		const {
+			highlightedItemIndex,
+			highlightedSectionIndex,
+			newHighlightedItemIndex,
+			newHighlightedSectionIndex,
+		} = params;
+		switch (event.which) {
+			case keycode.codes.esc:
+				event.preventDefault();
+				this.resetValue();
+				break;
+			case keycode.codes.enter:
+				if (!this.state.groups) {
+					break;
+				}
+				event.preventDefault();
+				if (Number.isInteger(highlightedItemIndex)) {
+					// suggestions are displayed and an item has the focus : we select it
+					this.onSelect(event, {
+						itemIndex: highlightedItemIndex,
+						sectionIndex: highlightedSectionIndex,
+					});
+				} else if (this.state.value !== this.state.previousValue) {
+					// there is no focused item and the current value is not persisted
+					// we persist it
+					this.updateValue(event, this.state.value, true);
+				}
+				// resetting suggestions to close dropdown
+				this.resetSuggestions();
+				break;
+			case keycode.codes.down:
+				event.preventDefault();
+				if (!this.state.groups) {
+					// display all suggestions when they are not displayed
+					this.updateGroups();
+					this.updateSelectedIndexes(this.state.value);
+					break;
+				}
+				this.setState({
+					focusedItemIndex: newHighlightedItemIndex,
+					focusedSectionIndex: newHighlightedSectionIndex,
+				});
+				break;
+			case keycode.codes.up:
+				event.preventDefault();
+				this.setState({
+					focusedItemIndex: newHighlightedItemIndex,
+					focusedSectionIndex: newHighlightedSectionIndex,
+				});
+				break;
+			default:
+				break;
+		}
+	}
+
+	/**
+	 * Select an item in suggestions list
+	 * @param event The select event
+	 * @param itemIndex The item index in suggestions list
+	 */
+	onSelect(event, { sectionIndex, itemIndex }) {
+		let newValue = this.state.groups[itemIndex];
+		if (this.props.multiSection) {
+			newValue = this.state.groups[sectionIndex].suggestions[itemIndex];
+		}
+		this.updateValue(event, newValue, true);
+	}
+
+	updateSelectedIndexes(value) {
+		if (this.props.multiSection) {
+			this.props.titleMap.forEach((group, sectionIndex) => {
```

Comment:
You may need to turn that into a `for` statement, to be able to break on first match